### PR TITLE
[formrecognizer] Add tests and samples for new features

### PR DESCRIPTION
**Pull request by catalinaperalta**
Part of #29774

### CODE REVIEW

File: sdk/formrecognizer/azure-ai-formrecognizer/samples/v3.2/async_samples/sample_analyze_invoices_async.py [hunk 1]
```diff
@@ -200,8 +200,8 @@ async def analyze_invoice_async():
             unit_price = item.value.get("UnitPrice")
             if unit_price:
                 print(
-                    "......Unit Price: {} has confidence: {}".format(
-                        unit_price.value, unit_price.confidence
+                    "......Unit Price: {}{} has confidence: {}".format(
+                        unit_price.value, unit_price.value.code, unit_price.confidence
```
Comment (kristapratico) [inline sdk/formrecognizer/azure-ai-formrecognizer/samples/v3.2/async_samples/sample_analyze_invoices_async.py:204]: if `code` is not returned by the service will this return empty string or None?

File: sdk/formrecognizer/azure-ai-formrecognizer/tests/test_dac_analyze_layout.py [hunk 2]
```diff
@@ -163,6 +163,34 @@ def test_layout_multipage_table_span_pdf(self, client):
...
+    @DocumentAnalysisClientPreparer()
+    @recorded_by_proxy
+    def test_layout_url_annotations(self, client):
+        poller = client.begin_analyze_document_from_url("prebuilt-layout", self.annotations_url_jpg)
+        layout = poller.result()
+        assert len(layout.pages) > 0
+        assert len(layout.pages[0].annotations) > 0
+        assert layout.pages[0].annotations[0].kind == "check"
+        assert layout.pages[0].annotations[0].polygon
+        assert layout.pages[0].annotations[0].confidence > 0.5
```
Comment (kristapratico) [inline sdk/formrecognizer/azure-ai-formrecognizer/tests/test_dac_analyze_layout.py:178]: should we just assert that `confidence` is not None? Or is `confidence` usually pretty stable with FR?

**Review by kristapratico — APPROVED** [review]

File: sdk/formrecognizer/azure-ai-formrecognizer/samples/v3.2/async_samples/sample_analyze_invoices_async.py (hunk 1 above)
Comment (catalinaperalta) [reply sdk/formrecognizer/azure-ai-formrecognizer/samples/v3.2/async_samples/sample_analyze_invoices_async.py]: We would be returning None. Just fixed how we print this. Thanks for pointing that out!

File: sdk/formrecognizer/azure-ai-formrecognizer/tests/test_dac_analyze_layout.py (hunk 2 above)
Comment (catalinaperalta) [reply sdk/formrecognizer/azure-ai-formrecognizer/tests/test_dac_analyze_layout.py:178]: Confidence does vary but this one is currently at around 0.7 I dont think it'll get close to 0.5 unless the service breaks a lot...so I think checking for a confidence above 0.5 should be fine

**Comment by catalinaperalta** [comment]
Live test pipeline run showing these tests passing: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=2697672&view=results

**Pull request merged**